frontend/src/ReportDataHandler.js

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from "react";
+import { useLoaderData } from "react-router";
+
+import App from "./App";
+import LoadingAnimation from "./components/LoadingAnimation";
+
+const ReportDataHandler = () => {
+  const [loading, setLoading] = useState(true);
+  let [lines, setLines] = useState([]);
+  const loaderData = useLoaderData("draftName");
+
+  useEffect(() => {
+    setLoading(true);
+    setLines(loaderData);
+
+    setLoading(false);
+  }, [loaderData]);
+
+  return <>{loading ? <LoadingAnimation /> : <App lines={lines} />}</>;
```

Comment:
We can't directly return JSX.

Reply:
What do you mean? React components should mainly return JSX. It could be something like this: `loading ? <LoadingAnimation /> : <App lines={lines} />` or even:
```js
if (loading) {
  return <LoadingAnimation/>
}
return <App ../>

```

Reply:
And BTW it looks like the loading screen only works on initial component mount. Maybe this hook can be useful for implementing loading screens? https://reactrouter.com/en/main/hooks/use-navigation

Reply:
> What do you mean?

Oops, sorry for the misspelling! I meant we can't use `javascript` directly in return statement that's why used Fragment.

> `loading ? <LoadingAnimation /> : <App lines={lines} />`

Oh! I see, I was using an extra return statement!

Reply:
> And BTW it looks like the loading screen only works on initial component mount...

Yes, working on it.

Reply:
> What do you mean?....

Sorry for the misspelling! I meant we can't directly use `return { ... }`. It would be considered an object.

> `loading ? <LoadingAnimation /> : <App lines={lines} />`

Oh! I see, you meant to remove `{...}` too and not just the fragment.

Reply:
> And BTW it looks like the loading screen only works on initial component mount.....

Fixed